src/Microsoft.AspNet.Authentication.OpenIdConnect/INonceCache.cs

```diff
@@ -5,8 +5,10 @@ namespace Microsoft.AspNet.Authentication.OpenIdConnect
 {
     public interface INonceCache
```

Comment:
Have you considered replacing this interface by `IDistributedCache`?

https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Framework.Caching.Distributed/IDistributedCache.cs